components/clarity-lsp/src/vscode_bridge.rs

```diff
@@ -22,10 +25,10 @@ use wasm_bindgen_futures::future_to_promise;
 
 #[wasm_bindgen]
 pub struct LspVscodeBridge {
-    editor_state_lock: Arc<RwLock<EditorState>>,
     client_diagnostic_tx: JsFunction,
     _client_notification_tx: JsFunction,
     backend_to_client_tx: JsFunction,
+    editor_state: EditorStateInput,
```

Comment:
What's your take on keeping the previous design here, and construct a `EditorStateInput::RwLock(self. editor_state_lock.clone())` when you send it to `process_notification` ?